### PR TITLE
Add SpringAnimation.js from React Native

### PR DESCRIPTION
**Pull request by Hixie**
Adding React Native spring animation code and license (as is). 
Part 1 of 2 per https://github.com/flutter/engine/pull/37604#issuecomment-1376536601

Issue: https://github.com/flutter/flutter/issues/105687#issuecomment-1314305754

See also: https://github.com/flutter/engine/pull/38733

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
The parts I didn't write are LGTM.
This will need a review from someone else for the parts I did write.

**Comment by Hixie** [comment]
test-exempt: this code as written does not execute yet. tests will come in a later PR.

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: third_party/spring_animation/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# spring_animation
```
Comment (zanderso) [inline third_party/spring_animation/README.md:1]: It's going to look like this dependency is unused. Should the README contain a warning against removing it?

Comment (Hixie) [reply third_party/spring_animation/README.md:1]: We'll be using it shortly.

**Pull request merged**